src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -262,6 +273,26 @@ public Optional<DetectBootResult> boot(String detectVersion, String detectBuildD
         return Optional.of(DetectBootResult.run(bootSingletons, propertyConfiguration, productRunData, directoryManager, diagnosticSystem));
     }
 
+    private BlackduckScanMode decideScanMode(BlackDuckConnectionDetails blackDuckConnectionDetails, Map<DetectTool, Set<String>> scanTypeEvidenceMap, boolean blackduckScanModeSpecified, DetectConfigurationFactory detectConfigurationFactory, boolean autonomousScanEnabled, DetectPropertyConfiguration detectConfiguration) {
+        if(!blackduckScanModeSpecified && autonomousScanEnabled) {
+            Optional<String> scaasFilePath = detectConfigurationFactory.getScaaasFilePath();
+            Optional<String> blackDuckUrl = blackDuckConnectionDetails.getBlackDuckUrl();
+
+            AllEnumList<DetectTool> detectTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS);
+
+            if (blackDuckUrl.isPresent()) {
+                boolean isNotRapid = detectTools.representedValues().stream().anyMatch(tool -> !rapidTools.contains(tool)) || scanTypeEvidenceMap.keySet().stream().anyMatch(tool -> !rapidTools.contains(tool));
+                if (!scanTypeEvidenceMap.isEmpty() && !isNotRapid && scaasFilePath.isPresent()) {
+                    return BlackduckScanMode.RAPID;
+                } else if (!scanTypeEvidenceMap.isEmpty() && scaasFilePath.isPresent()) {
+                    return BlackduckScanMode.STATELESS;
+                }
+            }
+            return BlackduckScanMode.INTELLIGENT;
+        }
+        return detectConfigurationFactory.createScanMode();
+    }
```

Comment:
Would be better to move to a ScanMode class.

Reply:
I think it is fine where it is now. Do not deserve its own class.